### ISSUE TITLE
Use Base.Math.rem_pio2_kernel for quadrant on Julia 0.7 

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalArithmetic.jl#119]
In Julia 0.7 we can use `Base.Math.rem_pio2_kernel` for the quadrant calculation for trig functions, especially `tan`.

(Thanks to Simon Byrne for the tip.)

**Comment by lbenet** [comment]
I was checking the way `mod2pi` is coded in Base (because `find_quadrants` is essentially a discrete version of `mod_pi_half` function; see #117). It turns out that `mod2pi(x)` is a call to `rem2pi(x, RoundDown)`; I wonder if this can be a work around that.

**Comment by dpsanders** [comment]
I tried that, but it's not good enough (or, at least, I couldn't get it to work) -- we actually need a `rem_half_pi` function, which is what the function in #119 provides.

**Comment by lbenet** [comment]
For floats, see what `ieee754_rem_pio2`; it is in Base.Math. I don't know how is `rem2pi` coded for `BigFloat`s, or whether there is something like `ieee754_rem_pio2` for them.

**Comment by dpsanders** [comment]
`rem2pi` just calls an MPFR function:

```
@edit rem(big"0.1", 2*BigFloat(pi), RoundNearest)
```

You're right that `ieee754_rem_pio2` would be the correct function, but:

```
julia> @btime Base.Math.ieee754_rem_pio2(3.1)
  39.199 ns (2 allocations: 128 bytes)
(2, [-0.0415927, 2.43541e-18])
```
whereas on 0.7:
```
julia> @btime Base.Math.rem_pio2_kernel(3.1)
  1.648 ns (0 allocations: 0 bytes)
(2, Base.Math.DoubleFloat64(-0.04159265358979315, 2.4354103626379128e-18))
```
!!

**Comment by dpsanders** [comment]
Fixed by #235.

**Comment by dpsanders** [comment]
Finally `rem2pi` was used.
